src/Files.App/Utils/Shell/ShellFileOperations2.cs

```diff
@@ -569,7 +569,7 @@ public void QueueRenameOperation(IEnumerable<ShellItem> sourceItems, string newN
 	/// <param name="disposing"><c>true</c> to release both managed and unmanaged resources; <c>false</c> to release only unmanaged resources.</param>
 	protected virtual void Dispose(bool disposing)
 	{
-		if (!disposedValue)
+		if (Interlocked.CompareExchange(ref disposedValue, 1, 0) == 0)
```

Comment:
How come this has to be thread safe? Just for my knowledge.

Reply:
If `Dispose` is called by multiple threads at the same time, the other thread will check `disposedValue` before it becomes true, and the disposal process may be done twice.

Reply:
Doesn't this class get instantiated in *a* STA thread?

Reply:
But `Dispose` is public and can be called from any thread. And if a NullReferenceException is thrown here, I can only assume it is because non-thread-safe.

Reply:
That makes sense. Thanks!!